lib/core/versions/latest/ProvisionalProofFile.ts

```diff
@@ -29,4 +46,68 @@ export default class ProvisionalProofFile {
 
     return compressedRawData;
   }
+
+  /**
+   * Parses and validates the given provisional proof file buffer.
+   * @param provisionalProofFileBuffer Compressed provisional proof file.
+   * @throws `SidetreeError` if failed parsing or validation.
+   */
+  public static async parse (provisionalProofFileBuffer: Buffer): Promise<ProvisionalProofFile> {
+    let provisionalProofFileDecompressedBuffer;
+    try {
+      const maxAllowedDecompressedSizeInBytes = ProtocolParameters.maxProofFileSizeInBytes * Compressor.estimatedDecompressionMultiplier;
+      provisionalProofFileDecompressedBuffer = await Compressor.decompress(provisionalProofFileBuffer, maxAllowedDecompressedSizeInBytes);
+    } catch (error) {
+      throw SidetreeError.createFromError(ErrorCode.ProvisionalProofFileDecompressionFailure, error);
+    }
+
+    let provisionalProofFileModel;
+    try {
+      provisionalProofFileModel = await JsonAsync.parse(provisionalProofFileDecompressedBuffer);
+    } catch (error) {
+      throw SidetreeError.createFromError(ErrorCode.ProvisionalProofFileNotJson, error);
+    }
+
+    if (provisionalProofFileModel.operations === undefined) {
+      throw new SidetreeError(ErrorCode.ProvisionalProofFileOperationsNotFound, `Provisional proof file does not have any operation proofs.`);
+    }
+
+    const operations = provisionalProofFileModel.operations;
+    const allowedProperties = new Set(['update']);
```

Comment:
Same for validateObjectContainsOnlyAllowedProperties

Reply:
Will do.